itest/lnd_payment_test.go

```diff
@@ -1351,3 +1353,44 @@ func runSendToRouteFailHTLCTimeout(ht *lntest.HarnessTest, restartAlice bool) {
 	ht.AssertPaymentStatus(alice, preimage.Hash(), lnrpc.Payment_FAILED)
 	ht.AssertPaymentStatusFromStream(payStream, lnrpc.Payment_FAILED)
 }
+
+// testSendPaymentKeysendMPPFail tests sending a keysend payment and trying to
+// split it will fail because keysend payment in combination with MPP is not
+// supported.
+func testSendPaymentKeysendMPPFail(ht *lntest.HarnessTest) {
+	const chanAmt = btcutil.Amount(100_000)
+	p := lntest.OpenChannelParams{Amt: chanAmt}
+
+	// Initialize the test context with 2 connected nodes.
+	cfgs := [][]string{nil, nil}
+
+	// Open and wait for channels.
+	_, nodes := ht.CreateSimpleNetwork(cfgs, p)
+	alice, bob := nodes[0], nodes[1]
+
+	// Send a keysend payment which has a total amount which is smaller
+	// than the maxShardAmt. The payment will still be recorded in the db
+	// but it will fail immediately because keysend and MPP cannot be
+	// combined.
+	dest, err := hex.DecodeString(bob.PubKeyStr)
+	require.NoError(ht, err)
+
+	// Create a preimage and corresponding payment hash for the keysend
+	// payment.
+	preimage := bytes.Repeat([]byte{0x10}, 32)
+	hash := sha256.Sum256(preimage)
+
+	client := alice.RPC.SendPayment(&routerrpc.SendPaymentRequest{
```

Comment:
I think we should let the `SendPayment` call fail, the fix should be at the RPC level as we can easily validate the param there without starting the more expensive payment flow.

Reply:
yes added an RPC level check instead.